lib/slim/embedded.rb

```diff
@@ -153,9 +153,11 @@ class SassEngine < TiltEngine
 
       def tilt_render(tilt_engine, tilt_options, text)
         text = tilt_engine.new(tilt_options.merge(
-          style: options[:pretty] ? :expanded : :compressed,
-          cache: false)) { text }.render
-        text = text.chomp
+          style: options[:pretty] ? :expanded : :compressed
+        ).merge!(
+          tilt_engine <= ::Tilt::SassTemplate && tilt_engine::Engine ? { cache: false } : { charset: false }
```

Comment:
This detects what the sass implementation is being used and use different option accordingly:
- `{ cache: false }` is used for `sassc` and `sass`.
- `{ charset: false }` is used for `sass-embedded`, this option ensures `@charset "UTF-8";` and BOM will not be emitted as they are both invalid sequence for inlined css.